src/WebJobs.Script/Environment/EnvironmentExtensions.cs

```diff
@@ -291,8 +291,17 @@ public static bool IsAppService(this IEnvironment environment)
         /// <returns><see cref="true"/> If running in a Kubernetes Azure App Service; otherwise, false.</returns>
         public static bool IsKubernetesManagedHosting(this IEnvironment environment)
         {
-            return !string.IsNullOrEmpty(environment.GetEnvironmentVariable(KubernetesServiceHost))
+            try
+            {
+                return !string.IsNullOrEmpty(environment.GetEnvironmentVariable(KubernetesServiceHost))
                 && !string.IsNullOrEmpty(environment.GetEnvironmentVariable(PodNamespace));
+            }
+            catch
+            {
+                //if there's an error reading the environment variable then return false
```

Comment:
I'd rather you just fix the test to mock things appropriately. Let's not have test code here.